### PR TITLE
Migrate remote provisioning toggle to ExtSettings

### PR DESCRIPTION
**Pull request by x86pup**

### CODE REVIEW

File: core/java/android/ext/settings/AttestRemoteConstants.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package android.ext.settings;
+
+/** @hide */
+public interface AttestRemoteConstants {
```
Comment (muhomorr) [inline core/java/android/ext/settings/AttestRemoteConstants.java]: I think `RemoteProvisioningConstants` would be a better name

File: core/java/android/ext/settings/AttestRemoteConstants.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    int GRAPHENEOS_ATTEST_SERVER = 0;
```
Comment (muhomorr) [inline core/java/android/ext/settings/AttestRemoteConstants.java]: `ATTEST` infix is not needed

File: core/java/android/ext/settings/AttestRemoteConstants.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    int STANDARD_ATTEST_SERVER = 1;
+    int DISABLED_ATTEST_SERVER = 2; // unused for now
```
Comment (muhomorr) [inline core/java/android/ext/settings/AttestRemoteConstants.java]: This is not a server, this is a future setting mode

File: core/java/android/ext/settings/ExtSettings.java [hunk 4]
```diff
@@ -31,6 +34,9 @@ public class ExtSettings {
     public static final BoolSetting SCRAMBLE_PIN_LAYOUT = new BoolSetting(
             Setting.Scope.PER_USER, "lockscreen_scramble_pin_layout", false);
 
+    public static final IntSetting ATTEST_REMOTE_PROVISIONER_SERVER = new IntSetting(
+        Setting.Scope.GLOBAL, "attest_remote_provisioner_server", GRAPHENEOS_ATTEST_SERVER);
```
Comment (muhomorr) [inline core/java/android/ext/settings/ExtSettings.java]: Valid setting values are missing

File: core/java/android/ext/settings/ExtSettings.java [hunk 5]
```diff
@@ -31,6 +34,12 @@ public class ExtSettings {
     public static final BoolSetting SCRAMBLE_PIN_LAYOUT = new BoolSetting(
             Setting.Scope.PER_USER, "lockscreen_scramble_pin_layout", false);
 
+    public static final IntSetting REMOTE_PROVISIONING_SERVER = new IntSetting(
+        Setting.Scope.GLOBAL, "attest_remote_provisioner_server", // historical setting key
+        GRAPHENEOS_PROXY, // default
+        STANDARD_SERVER, GRAPHENEOS_PROXY, DISABLED // valid values
```
Comment (muhomorr) [inline core/java/android/ext/settings/ExtSettings.java]: `DISABLED` is not a valid value currently.

Comment (x86pup) [reply core/java/android/ext/settings/ExtSettings.java]: Removed `DISABLED` from valid values

**Review by muhomorr — APPROVED** [review]

**Pull request merged**